src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -136,7 +142,13 @@ class BatchNorm
   OutputDataType TrainingMean() { return runningMean; }
 
   //! Get the variance over the training data.
-  OutputDataType TrainingVariance() { return runningVariance / count; }
+  OutputDataType TrainingVariance()
+  {
+    if (average)
+      return count ? runningVariance / count : runningVariance;
+
+    return runningVariance;
+  }
```

Comment:
If the training has not started yet we could log an error message saying that the training has not started yet.

Reply:
I think we should do that only where we take average, otherwise we won't be able to set running mean and variance if we directly want to run. Kindly let me know what you think.

Reply:
alright I think we can leave it like this for now.

Reply:
Great.